eth/db/storage.py

```diff
@@ -53,11 +67,18 @@ class StorageLookup(BaseDB):
 
     def __init__(self, db: DatabaseAPI, storage_root: Hash32, address: Address) -> None:
         self._db = db
+
+        # the root hash at the beginning of the block
         self._starting_root_hash = storage_root
+
         self._address = address
-        self._write_trie = None
+        self._write_trie: HexaryTrie = None
         self._trie_nodes_batch: BatchDB = None
 
+        # When deleting an account, push the pending write info onto this stack.
+        # May happen multiple times for multiple deletes.
+        self._historical_write_tries: List[Tuple[HexaryTrie, BatchDB, Hash32]] = []
```

Comment:
Maybe make a `NamedTuple` for the inner value type here.  Probably makes some of the other code a bit easier to read/grok.